### PR TITLE
fix: Fix context parsing

### PR DESCRIPTION
**Pull request by mahatoankitkumar**
## Problem
For number in context, it would consider it as an integer as whole, even if it was an array

## Solution
Fixed the parsing for number

### CODE REVIEW

**Review by sauraww — APPROVED** [review]

File: crates/frontend/src/utils.rs
Comment (ShubhranshuSanjeev) [inline crates/frontend/src/utils.rs:1]: This could be the case for other types also, like string.

**Review by ShubhranshuSanjeev — CHANGES_REQUESTED** [review]

File: crates/frontend/src/utils.rs [hunk 1]
```diff
@@ -338,21 +338,22 @@ pub fn parse_value(val: &str, config_type: ConfigValueType) -> Result<Value, Str
...
+                Value::Number(num) => Ok(Value::Number(num)),
+                Value::Array(arr) => {
+                    for item in &arr {
+                        if !item.is_number() {
+                            return Err("Array contains non-number value".to_string());
+                        }
+                    }
+                    Ok(Value::Array(arr))
+                }
+                _ => Err("Invalid number or number array format".to_string()),
```
Comment (Datron) [inline crates/frontend/src/utils.rs:354]: Can we also include the value that failed?

File: crates/frontend/src/utils.rs [hunk 2]
```diff
@@ -338,21 +338,22 @@ pub fn parse_value(val: &str, config_type: ConfigValueType) -> Result<Value, Str
...
-            .map(|number| Value::Number(number.into()))
-            .or_else(|_| {
-                f64::from_str(val)
-                    .ok()
-                    .and_then(|num| Number::from_f64(num).map(Value::Number))
-                    .ok_or_else(|| {
-                        "Invalid decimal format or precision issue".to_string()
-                    })
-            }),
+        ConfigValueType::Number | ConfigValueType::Integer => {
```
Comment (Datron) [inline crates/frontend/src/utils.rs:341]: We should treat these two differently, and parse them into their corresponding types. 
Integer - i32
Number - f32

Comment (Datron) [reply crates/frontend/src/utils.rs:341]: Open to opinions on this

File: crates/frontend/src/utils.rs [hunk 3]
```diff
@@ -338,21 +338,22 @@ pub fn parse_value(val: &str, config_type: ConfigValueType) -> Result<Value, Str
...
+                        if !item.is_number() {
+                            return Err("Array contains non-number value".to_string());
+                        }
+                    }
+                    Ok(Value::Array(arr))
+                }
+                _ => Err("Invalid number or number array format".to_string()),
+            }
+        }
         ConfigValueType::String => Ok(Value::String(val.to_string())),
```
Comment (pratikmishra356) [inline crates/frontend/src/utils.rs:357]: array check should be for string also

**Review by ShubhranshuSanjeev — APPROVED** [review]

**Review by pratikmishra356 — APPROVED** [review]

**Pull request merged**